### PR TITLE
Chore/ CI & CD for backend and executor service

### PR DESCRIPTION
**Pull request by limanjun99**
- Configure serverless for the /backend and /services/executor folders
- Setup Github actions to automatically test each pull request creation/update
- Allow different branches to deploy to dev environment by running the deploy-`service`-dev workflow

Next steps:
- Do the same for the frontend folder
- Create a new IAM user for production usage, and setup CI/CD for main branch

### CODE REVIEW

**Comment by limanjun99** [comment]
This PR is done, needs to be merged first (?) in order to use the workflows created here

**Comment by limanjun99** [comment]
.deploy

**Comment by limanjun99** [comment]
.deploy

**Comment by limanjun99** [comment]
.deploy

**Comment by limanjun99** [comment]
.deploy

**Comment by limanjun99** [comment]
.deploy

**Comment by limanjun99** [comment]
.deploy

**Comment by limanjun99** [comment]
.deploy

File: .github/workflows/branch-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+name: Test and deploy a branch to dev environment
```
Comment (huajun07) [inline .github/workflows/branch-deploy.yml:1]: Minor nitpick: Should change name

File: .github/workflows/branch-test.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+name: Test a pull request
```
Comment (huajun07) [inline .github/workflows/branch-test.yml:1]: Minor nitpick: Should change name

**Review by huajun07 — APPROVED** [review]
LGTM 
Approving again for the separating tests and deployment

**Pull request merged**